### DISCUSSION TITLE
How to isolate a local application completely (intercept inward and outward traffic)?

### DISCUSSION THREAD

**Discussion by avingoyal** [mitmproxy/mitmproxy#5739] [Intercept Everything!]
I'm using mitmproxy on my macOS machine. The requirement is simple - I would like to isolate a local application (java or node) completely. All the inward and outward traffic to/from this application should go through mitmproxy. 

I tried looking for answers but couldn't find one with concrete example or explanation. Can someone help here?

**Comment by b-onc** [comment]
https://mitmproxy.org/posts/local-redirect/macos/

You can use `--mode local` and even specify a specific application, the link has examples.

Alternatively, you can start mitmproxy in it's default mode (proxy at 8080) and configure the application to use this proxy (127.0.0.1:8080)
